### DISCUSSION TITLE
File Upload always uses validation in config file settings instead of class defined validation

### DISCUSSION THREAD

**Discussion by christoferd** [livewire/livewire#8103] [7. Bugs (archived)]
### Livewire version

3.4.7

### Laravel version

10.47

### Which PHP version are you using?

PHP 8.2.16

### Steps To Reproduce

1. Pull repo
2. Composer Install
3. Npm install
4. Setup DB "laravel"
5. edit .env DB user and password
6. php artisan migrate
7. in a terminal: npm run dev
8. in another terminal: php artisan serve
9. open in browser http://127.0.0.1:8000
10. click [ Choose File ] button and browse to a file that's _**not**_ an image
* here notice that Livewire did not use the validation specified above the class property
11. click the [ Upload ] button
* here also notice that Livewire did not use the validation specified above the class property

### Repository or Wirebox URL

https://github.com/christoferd/laravel-10-livewire-3-file-upload-validation-demo.git

### Additional information

I don't know if this is a bug, File Upload always uses validation in config file settings instead of class defined validation above the class property e.g.:
```php
#[Validate('file|max:12288|mimes:png,jpg,jpeg,webp', as: 'Archivo', message: 'Tipos de archivos aceptados: png/jpg/jpeg/webp. Tamaño máximo de archivo: 12.000 KB, o 12 MB')]
public $uploadedFile;
```

When calling ```$this->validate()``` I expect it to use the class property setting but instead I get // Exception Error: Missing [$rules/rules()] property/method on: [App\Livewire\ModelFileUpload].

I'm following the documentation from here:
https://livewire.laravel.com/docs/uploads#storing-uploaded-files
and
https://livewire.laravel.com/docs/validation#customizing-error-messages

**I don't see the documentation handling validation on preview, before upload.**

Perhaps the issue is that I am not using the store method on the uploaded file property, like: ```$photo->store(path: 'photos');```

Livewire is always validating the upload with the rules in the config file. 

I should be able to define different file upload validation rules for different classes, right?
I should be able to use the special validation definition code as shown in the documentation, right?

Environment ......................................................................................................................................
  Application Name ......................................................................................................................... Laravel
  Laravel Version .......................................................................................................................... 10.47.0
  PHP Version ............................................................................................................................... 8.2.16
  Composer Version .......................................................................................................................... 2.2.21
  Environment ................................................................................................................................ local
  Debug Mode ............................................................................................................................... ENABLED
  URL ............................................................................................................................... 127.0.0.1:8000
  Maintenance Mode ............................................................................................................................. OFF

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by joshhanley** [comment]
@christoferd due to Livewire side-loading files, it uploads the file to the server before the component is run and hence before validation on the component is run.

To ensure this process is still secure, there is some basic validation applied when a file is sideloaded, which can be updated in the Livewire config.

If that passes, then when you call `$this->validate()` in the component, then your component rules will function. But you will need to ensure that the temporary upload file rules have been changed to allow the correct files through.

Hope this helps!

**Comment by christoferd** [comment]
@joshhanley Thanks for the info, but doesn't explain Step #11.

Please note under Steps to Reproduce:
11. click the [ Upload ] button
here also notice that Livewire did not use the validation specified above the class property

**Reply by joshhanley** [reply]
@christoferd hmm ok not sure what is happening there. Any chance you could make a https://wirebox.app (make sure to make it public) so it's easier for us to have a look? Thanks!
